### ISSUE TITLE
Package request: Emudeck

### ISSUE THREAD

**Issue by ShalokShalom** [NixOS/nixpkgs#282130]
**Project description**

Emudeck is a program that abstracts video game emulators and optimises them for mobile game consoles like the Steam Deck. 

**Metadata**

* homepage URL: https://www.emudeck.com/
* source URL: https://github.com/dragoonDorise/EmuDeck
* license: mpl 2
* platforms: unix, linux, darwin, linux

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by Turmaxx** [comment]
After looking at the [emudeck wiki](https://emudeck.github.io), this isn't a package within itself, but rather a bunch of scripts that install, optimize and organize existing emulators. I am not sure how this will be implemented without wrapping or overriding with existing emulators found in nixpkgs. But seems like a nifty app though like retroarch.

**Comment by lpchaim** [comment]
Yeah, their approach sounds extremely tough to fit into the nix way of doing things, if it's feasible at all. 

In the meantime for the folks who stumble upon this, [RetroDeck](https://retrodeck.net/) might be worth a shot. It's a WIP alternative of sorts which just distributes a flatpak. I haven't personally tested it, but I'm assuming it'll work fine.

**Comment by dragoonDorise** [comment]
Hiya, EmuDeck creator over here :)

We do have an AppImage, would that work for you?

**Comment by Turmaxx** [comment]
Hey @dragoonDorise 

Glad you are here and got the time to check this issue out.

We are looking for a nix solution to add to the nixpkgs repository instead of the AppImage. I checked out the AppImage in the EmuDeck repository and I don't think it will be able to detect already existing emulator that were installed before hand on the system.

What I mean when I say the **nix** way of doing things is basically the same way that [RetroArch](https://nixos.wiki/wiki/RetroArch) was implemented. The emulators already exist in the nixpkgs repoistory they are just wrapped under the application, basically packaging them together. 

I am not sure if you are familiar with nix but reproducibility and declarative code is two of its key aspects and the code below with can be found in the link that I've provided above.

```nix
environment.systemPackages = with pkgs; [
  (retroarch.override {
    cores = with libretro; [
      genesis-plus-gx
      snes9x
      beetle-psx-hw
    ];
  })
];
```
RetroArch basically wraps the emulators and packages them together, then when you start RetroArch the emulators are there.
